Assets/MixedRealityToolkit-SDK/Features/Audio/Influencers/AudioInfluencerController.cs

```diff
@@ -0,0 +1,297 @@
+        }
+
+        /// <summary>
+        /// Maximum distance, in meters, to look when attempting to find the user and any influencers.
+        /// </summary>
+        /// <remarks>
+        /// The MaxDistance range is 1.0 to 50.0, inclusive.
+        /// The default value is 20.0.
+        /// </remarks>
+        [Tooltip("Maximum distance, in meters, to look when attempting to find the user and any influencers.")]
+        [Range(1.0f, 50.0f)]
+        [SerializeField]
+        private float maxDistance = 20.0f;
+        public float MaxDistance
+        {
+            get { return maxDistance; }
+            set
+            {
+                // set maxDistance and enforce the specified range
+                if (value < 1.0f)
+                {
+                    maxDistance = 1.0f;
+                }
+                else if (value > 50.0f)
+                {
+                    maxDistance = 50.0f;
+                }
+                else
+                {
+                    maxDistance = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Maximum number of objects that will be considered when looking for influencers. 
+        /// Setting this value too high may have a negative impact on the performance of your experience.
+        /// </summary>
+        /// <remarks>
+        /// MaxObjects can only be set in the Unity Inspector.
+        /// The MaxObjects range is 1 to 25, inclusive.
+        /// The default value is 10.
+        /// </remarks>
+        [Tooltip("Maximum number of objects that will be considered when looking for influencers.")]
+        [Range(1, 25)]
+        [SerializeField]
+        private int MaxObjects = 10;
+
+        /// <summary>
+        /// Time of last audio processing update. 
+        /// </summary>
+        private DateTime lastUpdate = DateTime.MinValue;
+
+        /// <summary>
+        /// The source of the audio.
+        /// </summary>
+        private AudioSource audioSource;
+
+        /// <summary>
+        /// The initial volume level of the audio source.
+        /// </summary>
+        private float initialAudioSourceVolume;
+
+        /// <summary>
+        /// The hits returned by Physics.RaycastAll
+        /// </summary>
+        private RaycastHit[] hits;
+
+        /// <summary>
+        /// The collection of previously applied audio influencers.
+        /// </summary>
+        private List<IAudioInfluencer> previousInfluencers = new List<IAudioInfluencer>();
+
+        /// <summary>
+        /// Potential effects manipulated by an audio influencer and their key
+        /// properties.
+        /// <summary>
+        private AudioLowPassFilter lowPassFilter;
+        private float nativeLowPassCutoffFrequency;
+        private AudioHighPassFilter highPassFilter;
+        private float nativeHighPassCutoffFrequency;
+
+        private void Awake() 
+        {           
+            audioSource = gameObject.GetComponent<AudioSource>();
```

Comment:
Per Unity docs, OnValidate is editor only. We need these values cached for the player as well

"This function is called when the script is loaded or a value is changed in the inspector (Called in the editor only).
Use this function to validate the data of your MonoBehaviours. This can be used to ensure that when you modify data in an editor that the data stays within a certain range."

Reply:
Right, but in the editor we would be serializing the value that the required component attached is. We can still check if it's null here before attempting to get it.

Reply:
As long as the field is serialized and doesn't reset itself, it will embed the reference in the scene/prefab. It's safe for some cases, and this looks like one of those cases.